pkg/controller/daemon/daemon_controller.go

```diff
@@ -865,12 +899,29 @@ func (dsc *DaemonSetsController) podsShouldBeOnNode(
 				continue
 			}
 			if pod.Status.Phase == v1.PodFailed {
+				failedPodsObserved++
+
+				// This is a critical place where DS is often fighting with kubelet that rejects pods.
+				// We need to avoid hot looping and backoff.
+				backoffKey := fmt.Sprintf("%s/%s", dsKey, node.Name)
+				delay := dsc.failedPodsBackoff.Get(backoffKey)
+
+				now := time.Now()
+				inBackoff := dsc.failedPodsBackoff.IsInBackOffSinceUpdate(backoffKey, now)
+				if inBackoff {
+					glog.V(4).Infof("Deleting failed pod %s/%s has been limited by backoff - %v remaining",
```

Comment:
if glog.V(4) {}

Reply:
maybe also add node name into debug

Reply:
`if glog.V(4) {}` my understanding is that this counts only if computing your arguments costs you

Adding nodeName seems useful, thx

Reply:
yeah, also if this will be called frequently you will make a lot of string allocations, but I guess it is fine here.